diff --git a/node_modules/react-native-safe-area-context/android/build.gradle b/node_modules/react-native-safe-area-context/android/build.gradle
index f324b9a..8f43947 100644
--- a/node_modules/react-native-safe-area-context/android/build.gradle
+++ b/node_modules/react-native-safe-area-context/android/build.gradle
@@ -1,15 +1,17 @@
 buildscript {
-    def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : project.properties['RNSAC_kotlinVersion']
+    if (project == rootProject) {
+        def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : project.properties['RNSAC_kotlinVersion']
 
-    repositories {
-        mavenCentral()
-        google()
-    }
+        repositories {
+            mavenCentral()
+            google()
+        }
 
-    dependencies {
-        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version")
-        classpath("com.android.tools.build:gradle:7.3.1")
-        classpath("com.diffplug.spotless:spotless-plugin-gradle:6.11.0")
+        dependencies {
+            classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version")
+            classpath("com.android.tools.build:gradle:7.3.1")
+            classpath("com.diffplug.spotless:spotless-plugin-gradle:6.11.0")
+        }
     }
 }
 
